gemini/sample-apps/fashion-trends-prediction/app/data_prep/driver.py

```diff
@@ -0,0 +1,332 @@
+    """Extracts data from Instagram and saves it in a JSON file.
+
+    Args:
+            scrape_size (dict): A dictionary containing the number of countries, influencers, and posts to scrape.
+            first_scrape (bool): A boolean indicating whether this is the first scrape.
+            saved (dict): A dictionary containing the saved data.
+
+    """
+
+    if first_scrape:
+        temp_map = {}
+        scraped_influencers = set()
+    else:
+        with open(data_path, "r") as f:
+            temp_map = json.load(f)
+        scraped_influencers = set(temp_map.keys())
+
+    cookies = instalogin()
+
+    for country_name in country_names[: scrape_size["num_countries"]]:
+        print(country_name)
+
+        # the top influencers of that country right now
+        influencers = getInfluencers(country_name)
+
+        for influencer in influencers[: scrape_size["num_influencers"]]:
+            print(influencer)
+
+            if (
+                influencer not in scraped_influencers
+            ):  # this influencer was not in the list of any other country till now
+                scraped_influencers.add(influencer)
+                # the influencer was there in the previous run
+                if influencer in saved["global"]:
+                    posts = getPosts(
+                        influencer,
+                        saved["global"][influencer],
+                        scrape_size["num_posts"],
+                        cookies,
+                        model="Gemini",
+                    )
+                else:
+                    posts = getPosts(
+                        influencer,
+                        [],
+                        scrape_size["num_posts"],
+                        cookies,
+                        model="Gemini",
+                    )
+
+                temp_map[influencer] = posts
+
+            time.sleep(1)
+
+            # save data after running for each influencer
+            with open(data_path, "w") as outfile:
+                json.dump(temp_map, outfile)
+
+    # delete the influencers that are not in the list of any country anymore
+    saved["global"] = {k: v for k, v in temp_map.items() if k in scraped_influencers}
+
+    with open(data_path, "w") as outfile:
+        json.dump(saved, outfile)
+
+
+def periodic_extraction(scrape_size, first_period):
+    """Extracts data from Instagram periodically and saves it in a JSON file.
+
+    Args:
+            scrape_size (dict): A dictionary containing the number of countries, influencers, and posts to scrape.
+            first_period (bool): A boolean indicating whether this is the first period.
+
+    """
+
+    if first_period:
+        saved = {}
+        saved["global"] = {}
+    else:
+        with open(data_path, "r") as f:
+            saved = json.load(f)
+
+    # first scrape is set to True when you are running for the first time in an extraction period
+    extract_json_influencer_wise(
```

Comment:
Is this extra function needed? Could it be combined into the current one?

Reply:
Added the code of this function in periodic_extraction itself.